### DISCUSSION TITLE
prelude-operator connection-string format

### DISCUSSION THREAD

**Discussion by j91321** [BishopFox/sliver#764] [Q&A] [answered]
Hi,

I'm a bit confused about how the `prelude-operator` command works. I have to provide `connection-string`, but I just couldn't find any information as what the actual format of it is. If I understand prelude API documentation you have to provide the token value to authenticate.

```
sliver > help prelude-operator connect

Connect with Prelude's Operator

Usage:
======
  connect [flags] connection-string

Args:
=====
  connection-string  string    connection string to the Operator Host

Flags:
======
  -a, --aes-key       string    AES key for communication encryption (default: abcdefghijklmnopqrstuvwxyz012345)
  -h, --help                    display help
  -r, --range         string    Agents range (default: sliver)
  -s, --skip-existing           Do not add existing sessions as Operator Agents
```

**Comment by rkervella** [comment, marked as answer]
The connection string should be the address of the TCP listener in prelude. If you're running the Sliver client and Operator on the same host, a valid example would be `prelude-operator connect 127.0.0.1:1234` (assuming prelude's TCP listener is listening on `1234/tcp`).
The way this integration works is the Sliver client registers itself as a new implant in Prelude for each Sliver session and beacon. We're not using the Prelude API for this at the moment (because it was non existent when we wrote this implementation).

**Reply by j91321** [reply]
Ah, that makes sense now. It would be worth adding it to the wiki or in the help message just for future reference. Anyway thanks for the answer.

**Reply by moloch--** [reply]
https://github.com/BishopFox/sliver/commit/3ec4c5129ce8a55d95973ab20cc8bdf02c0bb90b
